# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsGlobalWindow.h
# Commit: 4bed3a0e2332
# Full Hash: 4bed3a0e2332f24ffe38f08e234c8898c42249ee
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2016-11-07 02:21:38
# Regressor Bug: 1290021
# File Overlap Count: 1
# Description:
#   Bug 1290021 - Implement a prototype version of Houdini "Worklets Level 1" spec - part 1 - WebIDL interface, r=smaug
# ==============================================================================

diff -r d84108f190ee -r 4bed3a0e2332 dom/base/nsGlobalWindow.h
--- a/dom/base/nsGlobalWindow.h	Sun Nov 06 11:51:33 2016 +1300
+++ b/dom/base/nsGlobalWindow.h	Sun Nov 06 09:53:34 2016 +0100
@@ -136,6 +136,7 @@
 #if defined(MOZ_WIDGET_ANDROID) || defined(MOZ_WIDGET_GONK)
 class WindowOrientationObserver;
 #endif
+class Worklet;
 namespace cache {
 class CacheStorage;
 } // namespace cache
@@ -926,6 +927,10 @@
   static void
   ConvertDialogOptions(const nsAString& aOptions, nsAString& aResult);
 
+  // Exposed only for testing
+  already_AddRefed<mozilla::dom::Worklet>
+  CreateWorklet(mozilla::ErrorResult& aRv);
+
 protected:
   bool AlertOrConfirm(bool aAlert, const nsAString& aMessage,
                       nsIPrincipal& aSubjectPrincipal,